File: src/main/java/com/axentx/surrogate1/logging/Logger.java
package com.axentx.surrogate1.logging;

import java.util.logging.Level;
import java.util.logging.Logger;

public class Logger {
    private static final java.util.logging.Logger logger = java.util.logging.Logger.getLogger(Logger.class.getName());

    public static void logCriticalEvent(String message) {
        logger.log(Level.SEVERE, "CRITICAL EVENT: " + message);
    }

    public static void logError(String message, Throwable throwable) {
        logger.log(Level.SEVERE, "ERROR: " + message, throwable);
    }

    public static void logInfo(String message) {
        logger.log(Level.INFO, "INFO: " + message);
    }

    public static void logWarning(String message) {
        logger.log(Level.WARNING, "WARNING: " + message);
    }
}

File: embedding/logger.py
import logging

class EmbeddingLogger:
    def __init__(self):
        self.logger = logging.getLogger('embedding_logger')
        self.logger.setLevel(logging.DEBUG)

        # Create handlers
        c_handler = logging.StreamHandler()
        f_handler = logging.FileHandler('/opt/axentx/surrogate-1/logs/embedding.log')
        c_handler.setLevel(logging.WARNING)
        f_handler.setLevel(logging.DEBUG)

        # Create formatters and add it to handlers
        c_format = logging.Formatter('%(name)s - %(levelname)s - %(message)s')
        f_format = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
        c_handler.setFormatter(c_format)
        f_handler.setFormatter(f_format)

        # Add handlers to the logger
        self.logger.addHandler(c_handler)
        self.logger.addHandler(f_handler)

    def log_success(self, message):
        self.logger.info(message)

    def log_error(self, message):
        self.logger.error(message)

    def log_debug(self, message):
        self.logger.debug(message)